packages/headless/src/composables/typeAhead.ts

```diff
@@ -0,0 +1,26 @@
+import { isPrintableCharacter } from "../utils/keyEvent";
+import { debounce } from "../utils/timer";
+
+/**
+ * Enhances typeahead to combine multiple inputs in quick succession and filter out non-printable characters.
+ *
+ * @example
+ * ```ts
+ * const typeAhead = useTypeAhead((inputString) => onTypeAhead(inputString));
+ * // ...
+ * const onKeydown = (key: KeyboardEvent) => typeAhead(key.event);
+ * ```
+ */
+export const useTypeAhead = (callback: (input: string) => void) => {
+  let inputString = "";
+  const debouncedReset = debounce(() => (inputString = ""), 500);
```

Comment:
maybe we should export the "500" as const and use it in the typeAhead.spec.ts and then add 100 in there, so if we change the debounce timeout for any reason, it will still work. (on the other hand the test will fail and we can adjust it then if needed.)

Or are we sure that 500 is the optimum and will not be touched?